### ISSUE TITLE
LC0003 False Positive: AccessByPermission already uses object name

### ISSUE THREAD

**Issue by FBakkensen** [StefanMaron/BusinessCentral.LinterCop#1173]
# LC0003 False Positive: AccessByPermission already uses object name

## Summary
LinterCop rule **LC0003** ("Do not use an Object ID for properties or variable declarations") is raised against a field whose `AccessByPermission` property already uses the table *name* `"Shipping Agent Services"` rather than a numeric object ID. The suggested fix in the diagnostic message is identical to the existing code, indicating a false positive or stale analysis state.

## Diagnostic Message (as shown)
```
app\src\Workflows\ShipmentLogistics\NALJPJobShipmentHeader.Table.al(625,34): error LC0003: Do not use an Object ID for properties or variables declaration. Use tabledata "Shipping Agent Services" = R instead.
```
(Note: Severity appeared as `error` in build output though rule default is `Warning`; may be escalated by configuration.)

## Actual Source (excerpt)
Field definition around the reported line (field 105):
```al
field(105; "Shipping Agent Code"; Code[10])
{
    AccessByPermission = tabledata "Shipping Agent Services" = R;
    Caption = 'Shipping Agent Code';
    TableRelation = "Shipping Agent";
    ToolTip = 'Specifies the shipping agent code.';
}
```
This already follows the rule (no numeric object ID).

Other nearby `AccessByPermission` usages also use names, e.g.:
```al
field(5792; "Shipping Time"; DateFormula)
{
    AccessByPermission = tabledata "Shipping Agent Services" = R;
    Caption = 'Shipping Time';
    AllowInCustomizations = Always;
    ToolTip = 'Specifies the shipping time calculation.';
}
field(5793; "Outbound Whse. Handling Time"; DateFormula)
{
    AccessByPermission = tabledata Location = R; // also name-based
    Caption = 'Outbound Whse. Handling Time';
}
```

## What Was Checked
Repository-wide searches (regex):
- `AccessByPermission = tabledata [0-9]` -> no matches
- `: Codeunit [0-9]` -> no matches
- `Record [0-9]` -> no matches
- No page/report/LookupPageId/DrillDownPageId properties with raw numeric IDs.

No remaining numeric object references appear in the project for these patterns, so LC0003 should not trigger here.

## Expected Behavior
No LC0003 diagnostic when the code already uses the object name form in `AccessByPermission`.

## Actual Behavior
LC0003 fires and proposes an identical replacement (`tabledata "Shipping Agent Services" = R`).

## Hypotheses
1. **Stale Analyzer Cache**: Previous version may have used numeric ID; analyzer not refreshed, but re-opening VS Code and rebuilding did not clear it.
2. **Span Mapping Issue**: Diagnostic attached to wrong updated line after file edits.
3. **Parser Edge Case**: Specific token sequence in this large table object confuses the rule.
4. **Severity Escalation**: Configuration transformed warning into error, but the underlying detection still misfired.

## Environment
| Item | Value |
|------|-------|
| App name | 9A Advanced Manufacturing - Project Based |
| App version | 26.9.0.0 |
| Runtime (app.json) | 15.0 |
| Platform (app.json) | 26.0.0.0 |
| Application (app.json) | 26.3.0.0 |
| LinterCop Rule | LC0003 |
| Field Object ID | 105 (field number inside table 71115682) |
| Table Object | `table 71115682 "NALJP Job Shipment Header"` |

(Exact AL Language extension & LinterCop extension versions were not programmatically collected in this report; can add if needed.)

## Repro Steps (Best Effort)
1. Have a large table object with multiple `AccessByPermission` properties using name-based tabledata references.
2. Build/compile with LinterCop enabled.
3. Observe LC0003 warning/error on a line already using the name.

## Additional Context
The rule test case for LC0003 flags numeric usage (e.g. `Codeunit 50100`). Our code mirrors the already-correct *NoDiagnostic* scenario, making this appear to be a false positive.

## Requested Action
- Confirm whether `AccessByPermission` handling includes a branch that still expects numeric -> name transformation and falsely reports even when already compliant.
- Provide guidance or patch if mis-detection is confirmed.

## Temporary Workaround Considered
Avoided adding `#pragma warning disable LC0003` until clarified to prevent masking real violations elsewhere.

## Attachments / Snippet
Included above; can supply full table file privately if needed.

Thanks for the great tool—happy to run experimental builds or add logging if you point to a diagnostic flag.

**Comment by Arthurvdv** [comment]
Hi, over the past months I’ve been putting a lot of time and care into the successor of BusinessCentral.LinterCop: ALCops (https://alcops.dev/). Because of that, this repository is no longer actively maintained.

I’m closing this issue, as it won’t be handled here anymore. If this is still relevant for you, please have a look at ALCops/Analyzers (https://github.com/ALCops/Analyzers).

To make the transition easier, you can also follow the LinterCop Migration Guide: https://alcops.dev/docs/lintercop-migration/

If you have any questions or run into issues during the transition, feel free to reach out through ALCops. Happy to help 🙂
